### PR TITLE
ENH: CSS only sidebar toggling

### PR DESCRIPTION
**Pull request by choldgraf**
This is updates our left sidebar behavior so that instead of using javascript functions to control the display of the left sidebar with a class, instead we use a CSS input box + `:checked` to toggle its visibility. This will make these pages a bit lighter and less-complex, and make them more robust if wonky things happen with javascript on the page.

It also implements Furo-style left sidebar behavior on mobile devices, so that the whole left sidebar pops up in an overlay and you close it by clicking elsewhere on the screen.

### Old behavior

(shown on widescreen first, then mobile) (sorry for the red tint)

RtDtZ49OZj

### New behavior

Note that the left TOC closes automatically, because it will be hidden by default on pageload on mobile

gFg5I9auA1

### Refs

closes https://github.com/executablebooks/sphinx-book-theme/issues/366
closes https://github.com/executablebooks/sphinx-book-theme/issues/355
closes https://github.com/executablebooks/jupyter-book/issues/1379
closes https://github.com/executablebooks/jupyter-book/issues/1361
closes https://github.com/executablebooks/jupyter-book/issues/950

### CODE REVIEW

**Comment by choldgraf** [comment]
Curious is @pradyunsg has any other suggestions here, or if this looks mostly reasonable to you :-)

cc also @AakashGfude since this restructures a little bit of the HTML

File: sphinx_book_theme/layout.html [hunk 1]
```diff
@@ -9,7 +9,12 @@
 
 <!-- Docs TOC is "d-none d-xl-block col-xl-2" -->
 {% block docs_sidebar %}
-<div class="col-12 {{ sidebar_width_class }} bd-sidebar site-navigation show{% if is_single_page %} single-page{% endif %}" id="site-navigation">
+<!-- Checkboxes to toggle the left sidebar -->
+<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
```
Comment (pradyunsg) [inline sphinx_book_theme/layout.html:13]: Add an aria-label into this?

Comment (pradyunsg) [reply sphinx_book_theme/layout.html:13]: Or a labelled-by?

**Review by pradyunsg — COMMENTED** [review]
LGTM overall!

**Pull request merged**